### DISCUSSION TITLE
rmarkdown support

### DISCUSSION THREAD

**Discussion by albersonmiranda** [chipzoller/hugo-clarity#183] [Q&A]
Hi!

I'm using this awesome theme to blog via R's package {blogdown} (see example [here](https://datamares.netlify.app/post/atestando-diferencas-em-medias-o-teste-t-para-amostras-independentes/)) but for some reason all code features won't work (folding, line numbers etc). My guess is  it occurs because of code block with rmarkdown are enclosed like

> \<pre class="r">\<code>
> \## code goes here ##
> \</code>"

What should I change in code.js, _syntax.sass or elsewhere to enable r class and get it working?

obs.: .md theme's examples renders just fine

**Comment by chipzoller** [comment]
This is not something we planned for by default, @albersonmiranda, so will require some input.

**Comment by onweru** [comment]
@albersonmiranda, 
Umh, the code block features are based on [hugo's default highlighting configurations](https://gohugo.io/getting-started/configuration-markup#highlight), which are based on goldmark markdown renderer. At this point, [we've limited our scope in fact](https://github.com/chipzoller/hugo-clarity/blob/b67849d1ca101457d245fef9e79a9325a8b52468/exampleSite/config.toml#L40). Our next step would be to support all functionality when this condition -> `lineNumbersInTable = true` is set. 

I, therefore, don't see myself working on rmarkdown support soon. I would, however, encourage you or anyone else to contribute in that direction.

**Comment by albersonmiranda** [comment]
Apparently [it's not possible](https://discourse.gohugo.io/t/using-highlight-with-rmarkdown/29307). Looks like it's a job for rmarkdown side.

**Comment by albersonmiranda** [comment]
Guys,

Do you see at least a way (via custom css I guess) to make `code` class inherit `pre` class? I think it'd allow highlight.js to work properly with rmarkdown. Right now it randomly guesses code (never correctly lol) as bash, yaml, ruby...

**Comment by onweru** [comment]
@albersonmiranda, I’ll look into it.

**Comment by albersonmiranda** [comment]
I don't know if it helps, but I managed to think of a workaround to enable highlight, copy, line numbers, block height etc:

```
# code to be shown in the blog
blogdown::shortcode("highlight", "r", .content = "
data = mtcars
head(data)
")

#code do be evaluated
data = mtcars
head(data)
```
Code encapsulated by `blogdown::shortcode()` returns
```
{{% highlight "r" %}}

data = mtcars        
head(data)

{{% /highlight %}}
```
Which makes everything render properly. Downside is that the code isn't evaluated so I have to duplicate everyline that has to be evaluated by R engine. You can see an example in the end of [this post](https://datamares.netlify.app/en/post/r-vscode/).

Hope this helps you guys to understand the issue and maybe figure out a solution (hopefully).
